XenAdmin/Dialogs/HealthCheck/HealthCheckOverviewDialog.cs

```diff
@@ -279,15 +279,26 @@ public void UpdateAnalysisResult(HealthCheckSettings healthCheckSettings)
                 previousUploadPanel.Visible = false;
             }
         }
-
+        
         public void UpdateButtonsVisibility(Pool pool)
         {
+            bool canAlterHealthCheck = Core.HealthCheck.PassedRbacChecks(pool.Connection);
+
             refreshLinkLabel.Visible =
                 editLinkLabel.Visible =
                 disableLinkLabel.Visible =
                 uploadRequestLinkLabel.Visible =
-                enrollNowLinkLabel.Visible = Core.HealthCheck.PassedRbacChecks(pool.Connection);
+                enrollNowLinkLabel.Visible = canAlterHealthCheck;
+
+            if (!canAlterHealthCheck)

```

Comment:
We need an else block (or better reverse the if-check block so the condition is easier to grasp) to hide the message in the opposite case  (compare what happens now on the panel if you are connected to one pool as an unprivileged user and to another pool as a privileged one and you select the second pool afer the first one).